cmd/vulcan-trivy/vulcan-trivy.go

```diff
@@ -0,0 +1,337 @@
+		log.Printf("Skipping cache update with a latest dockerimage: %s\n", target)
+		triviArgs = append(triviArgs, "--skip-update")
+	}
+	// Show only vulnerabilities with fixes.
+	if opt.IgnoreUnfixed {
+		triviArgs = append(triviArgs, "--ignore-unfixed")
+	}
+	// Show only vulnerabilities with specific severities.
+	if opt.Severities != "" {
+		severitiesFlag := []string{"--severity", opt.Severities}
+		triviArgs = append(triviArgs, severitiesFlag...)
+	}
+	// Append the target (docker image including registry hostname).
+	triviArgs = append(triviArgs, target)
+
+	log.Printf("Running command: %s %s\n", triviCmd, triviArgs)
+
+	err := retry.Do(
+		func() error {
+			cmd := exec.Command(triviCmd, triviArgs...)
+			cmdOutput, err := cmd.CombinedOutput()
+			if err != nil {
+				log.Printf("exec.Command() failed with %s\nCommand output: %s\n", err, string(cmdOutput))
+				return errors.New("trivy command execution failed")
+			}
+			log.Printf("trivy command execution completed successfully")
+			return nil
+		},
+		retry.Attempts(3),
+		retry.DelayType(retry.RandomDelay),
+		retry.MaxJitter(5*time.Second),
+	)
+	if err != nil {
+		log.Printf("retry exec.Command() failed with error: %s\n", err)
+		return errors.New("trivy command execution failed")
+	}
+
+	jsonFile, err := os.Open(reportOutputFile)
+	if err != nil {
+		log.Printf("trivy report output file read failed with error: %s\n", err)
+		return errors.New("trivy report output file open failed")
+	}
+	log.Printf("successfully open trivy report output file %s\n", reportOutputFile)
+	defer jsonFile.Close()
+
+	byteValue, err := ioutil.ReadAll(jsonFile)
+	if err != nil {
+		log.Printf("trivy report output file read failed with error: %s\n", err)
+		return errors.New("trivy report output file read failed")
+	}
+
+	var results ScanResponse
+	err = json.Unmarshal(byteValue, &results)
+	if err != nil {
+		return errors.New("unmarshal trivy output failed")
+	}
+
+	// If there are no vulnerabilities we can return.
+	if len(results) < 1 {
+		return nil
+	}
+
+	ap := report.ResourcesGroup{
+		Name: "Affected Packages",
+		Header: []string{
+			"Name",
+			"Version",
+			"Severity",
+			"FixedBy",
+		},
+	}
+
+	vp := report.ResourcesGroup{
+		Name: "Package Vulnerabilities",
+		Header: []string{
+			"Name",
+			"Version",
+			"Vulnerabilities",
+		},
+	}
+
+	var rows []map[string]string
+	duppedPackageVulns := make(map[string]map[string]string)
+	// As we are scanning only one container per check, we have only one item
+	// in results array.
+	for _, feature := range results[0].Vulnerabilities {
+		// Set global score for the report.
+		score := getScore(feature.Severity)
+		if score > vuln.Score {
+			vuln.Score = score
+		}
+
+		ap := map[string]string{
+			"Name":            feature.PkgName,
+			"Version":         feature.InstalledVersion,
+			"Severity":        feature.Severity,
+			"FixedBy":         feature.FixedVersion,
+			"Vulnerabilities": fmt.Sprintf("%s", feature.Description),
```

Comment:
Now this is more the description of one vulnerability than a list of vulnerabilities, right? Maybe we can change this field to be a set of `VulnerabilityID` from the `feature`. Then if we want to keep a description because looks more cool, maybe we can add a new field.

Reply:
Absolutely agree. Changed description for VulnerabilityIDs + link.
Truncated the number of VulnerabilityIDs to 10 to avoid overflow the report.